src-self-hosted/translate_c.zig

```diff
@@ -815,6 +815,9 @@ fn transRecordDecl(c: *Context, record_decl: *const ZigClangRecordDecl) Error!?*
                 raw_name = try std.fmt.allocPrint(c.a(), "unnamed_{}", .{c.getMangle()});
                 is_anon = true;
             }
+            if (raw_name.len == 0) {
```

Comment:
On msvc
```c
typedef struct NAMED
{
    long name;
} NAMED;

typedef struct ONENAMEWITHSTRUCT
{
    NAMED;
    long b;
} ONENAMEWITHSTRUCT;
```
should generate
```zig
pub const struct_NAMED = extern struct {
    name: c_long,
};
pub const NAMED = struct_NAMED;
pub const struct_ONENAMEWITHSTRUCT = extern struct {
    unnamed_1: struct_NAMED,
    b: c_long,
};
```
which should happen if you just move `or raw_name.len == 0` to line 814.

You can add a test for this by using:
```zig
if (std.Target.current.abi == .msvc) {
    ...
} else {
    ...
}

Reply:
Sounds good!